common/src/main/java/com/wynntils/screens/maps/widgets/PoiManagerWidget.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.maps.widgets;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.components.Managers;
+import com.wynntils.features.map.MapFeature;
+import com.wynntils.models.map.pois.CustomPoi;
+import com.wynntils.screens.maps.PoiCreationScreen;
+import com.wynntils.screens.maps.PoiManagementScreen;
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.mc.RenderedStringUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import java.util.List;
+import java.util.Optional;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.network.chat.Component;
+
+public class PoiManagerWidget extends AbstractWidget {
+    private CustomPoi poi;
+    private Button editButton;
+    private Button deleteButton;
+    private Button upButton;
+    private Button downButton;
+    private int row;
+    private CustomColor color;
+    private int spacingMultiplier = 20;
+    private PoiManagementScreen managementScreen;
+    private List<CustomPoi> pois;
+
+    public PoiManagerWidget(
+            float x, float y, int width, int height, CustomPoi poi, int row, PoiManagementScreen managementScreen) {
+        super((int) x, (int) y, width, height, Component.literal(poi.getName()));
+        this.poi = poi;
+        this.row = row;
+        this.managementScreen = managementScreen;
+
+        pois = Managers.Feature.getFeatureInstance(MapFeature.class).customPois.get();
+
+        color = CustomColor.fromInt(0xFFFFFF);
+
+        if (poi.getVisibility() == CustomPoi.Visibility.HIDDEN) {
+            color = CustomColor.fromInt(0x636363);
+        }
+
+        this.editButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.poiManagementGui.edit"),
+                        (button) -> McUtils.mc().setScreen(PoiCreationScreen.create(managementScreen, poi)))
+                .pos(this.width / 2 + 85 + 20, 54 + spacingMultiplier * row)
+                .size(40, 20)
+                .build();
+
+        this.deleteButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.poiManagementGui.delete"), (button) -> {
+                            Managers.Feature.getFeatureInstance(MapFeature.class)
```

Comment:
Having a delete without either a confirmation, or a way to undo, is a bit unforgiving.

(I'd prefer a "Undo last delete" rather than a confirmation)

Reply:
I think that's overkill. Look at how Github does when you delete a branch after merging a PR. I think that's pretty nifty. You can delete it with just a single click, but if you mis-clicked, you get the chance to undo. So maybe do something like this:
when deleting a poi, save it as "last deleted" in the screen, and in the screen, if there is a "last deleted" poi, show a "undo delete" button (which is otherwise hidden, or possibly visible but greyed out), and if that is pressed, just re-add the poi. And don't bother trying to persist the "last deleted poi", if you exit the screen well then you're out of luck.

I think that'd work nicely.

Reply:
I've added the undo last delete button, but it only stores 1 deletion, do you think it would be a good idea to instead store a list of deleted pois so then you can undo everything you've deleted in the current "session".

Reply:
No, I think that is enough. It's a bit of a safety, but we're not overdoing it.